tox.ini

```diff
@@ -50,6 +50,11 @@ basepython = python3.10
 deps = -r{toxinidir}/test-requirements.txt
 commands = nose2 {posargs} --with-coverage -s tests/
 
+[testenv:py311]
```

Comment:
add it to the gh workflow matrix, please - https://github.com/juju/charm-helpers/blob/master/.github/workflows/build.yml#L18

Reply:
I deliberately didn't do that for this one; py3.11 fails, and I just wanted to make the lib work with py3.5 again.  It's deliberately a smaller PR; we can tackle the breaking py3.11 issues in another PR.

Reply:
ack.